### ISSUE TITLE
Incorrect cursor used on buttons

### ISSUE THREAD

**Issue by karlgroves** [WordPress/gutenberg#15343]
Incorrect cursor used on buttons

-   **Severity:** Low
-   **Affected Populations:**
    -   Low-Vision
    -   Cognitively Impaired
-   **Platform(s):**
    -   All / Universal
-   **Components affected:**
    -   Global

#### Issue description

Buttons throughout the application use the pointer cursor instead of the
default.

The use of consistent and relevant cursors may be important for users
who have a cognitive disability, since cursors give a visual clue as to
an element's functionality. Using the pointer cursor for elements which
do not typically show that cursor may be confusing or counter-intuitive
for users.

##### Issue Code

```
    label {


        cursor: pointer;


    }





    .wp-core-ui .button,


    .wp-core-ui .button-primary,


    .wp-core-ui .button-secondary {


        cursor: pointer;


        ...


    }





    input[type=checkbox],


    input[type=radio] {


        cursor: pointer;


        ...


    }





    .components-button {


       cursor: pointer;


        ...


    }

```

#### Remediation Guidance

Only controls that navigate users to new pages/views should have the
cursor set to `pointer`, while controls performing push-button type
actions use `default`.

##### Recommended Code

```
    label {


        cursor: default;


    }





    .wp-core-ui .button,


    .wp-core-ui .button-primary,


    .wp-core-ui .button-secondary {


        cursor: default;


        ...


    }





    input[type=checkbox],


    input[type=radio] {


        cursor: default;


        ...


    }





    .components-button {


       cursor: default;


        ...


    }
```

#### References

<https://www.w3.org/TR/CSS22/ui.html#cursor-props>

**Note:** This issue may be a duplicate with other existing accessibility-related bugs in this project. This issue comes from the Gutenberg accessibility audit, performed by [Tenon](https://www.tenon.io) and funded by [WP Campus](https://wpcampus.org/). This issue is GUT-9 in Tenon's report

**Comment by afercia** [comment]
Worth noting this is a "feature" used across all the WordPress admin. Gutenberg uses its own `cursor: pointer` rule, and in other cases the same property is inherited by the core CSS.

**Comment by melchoyce** [comment]
Screenshots from full report:

![image](https://user-images.githubusercontent.com/2846578/57185725-74281380-6e85-11e9-9834-56c0c6507e1b.png)

![image](https://user-images.githubusercontent.com/2846578/57185727-7a1df480-6e85-11e9-8ce9-c8a68be9d7a3.png)

**Comment by afercia** [comment]
Same for all the `<label>` elements and buttons in core:

<img width="620" alt="Screenshot 2019-05-05 at 14 08 46" src="https://user-images.githubusercontent.com/1682452/57193853-14427300-6f40-11e9-8c3c-5420d5bbf057.png">

<img width="374" alt="Screenshot 2019-05-05 at 14 08 54" src="https://user-images.githubusercontent.com/1682452/57193854-14db0980-6f40-11e9-8b53-88ee2141b128.png">

**Comment by getdave** [comment]
Does this need changing or are we sticking with the pattern established in WP Core?

**Comment by afercia** [comment]
For full compliance, it should be changed. I guess it needs be discussed by the accessibility and design teams.

**Comment by kjellr** [comment]
> Only controls that navigate users to new pages/views should have the
cursor set to pointer, while controls performing push-button type
actions use default.

That's interesting! I've never known this was specifically against a11y guidelines. In general on the web, I tend to expect that most "clickable" elements use a pointer cursor (take the buttons here on GitHub for instance, which have a `cursor: pointer` rule). I'd love to have more discussion around this.

**Comment by afercia** [comment]
Interesting issue, indeed. I don't think this is explicitly mentioned anywhere in the WCAG (I may be wrong). However, it has more to do with operating systems / browsers default: they shouldn't be changed unless there are very, very, good reasons to do so.

The native cursor style for buttons is `default`: https://codepen.io/afercia/full/EzjXMo

The W3C spec (CSS 2.2) linked above clearly states:
> pointer: The cursor is a pointer that indicates a link.

Same for CSS3 UI: https://www.w3.org/TR/css-ui-3/#cursor

The rationale is that `pointer` should be reserved for links, to help users differentiate UI controls that trigger navigation from buttons and the like. Granted, in WordPress you never know if what are you going to click is a button or a link 🙂For historical reasons, sometimes links are styled as buttons, and buttons are styled as links. Ideally, this shouldn't happen. Links should be clearly identified as links. Buttons should be identified as buttons.

After some software archeology, turns out some of the related changes are very _ancient_:

`<label>` elements got their `cursor: pointer` 15 years ago in:
https://core.trac.wordpress.org/changeset/560
no related Trac ticket.

`<button>` elements 11 years ago in:
https://core.trac.wordpress.org/changeset/7215
related Trac ticket:
https://core.trac.wordpress.org/ticket/5943

**Comment by enriquesanchez** [comment]
Discussed today during the design's team Gutenberg triage [in Slack](https://core.trac.wordpress.org/ticket/47171).

Created Trac ticket: https://core.trac.wordpress.org/ticket/47171 for follow up.

**Comment by mapk** [comment]
I'm going to close this in favor of a more holistic Core approach. Thanks for creating a [ticket in Trac](https://core.trac.wordpress.org/ticket/47171), @enriquesanchez!
